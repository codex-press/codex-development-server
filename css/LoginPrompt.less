
.LoginPrompt {
  position: fixed;
  width: 400px;
  left: calc(50% - 200px);
  top: 20%;
  padding: 1em;
  font-size: 1.2rem;

  &:extend(.modal);

  h3 {
    font-size: 1.4rem;
    margin: 0;
  }

  label {
    display: block;
    margin: 1rem 0;
  }

  input {
    width: 100%;
    font-size: 1.5rem;
  }

  &.invalid input {
    color: red;
    outline: 1px solid red;
  }

  .button-container {
    text-align: right;
  }

}


