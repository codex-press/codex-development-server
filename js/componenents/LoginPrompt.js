import React from 'react'

import { api } from '../utility'


export default class LoginPrompt extends React.Component {

  constructor() {
    super();
    this.state = {};
  }


  render() {
    return (
      <form 
        className={ 'LoginPrompt' + (this.state.invalid ? ' invalid' : '') }
        onSubmit ={ e => this.submit(e) }>

        <h3>Codex Press Development Server</h3>

        <label>
          Email
          <input type="email" />
        </label>

        <label>
          Password
          <input type="password" />
        </label>

        <div className="button-container">
          <button type="submit" className="primary">
            Login
          </button>
        </div>

      </form>
    );
  }


  submit(e) {
    e.preventDefault();
    let email = e.target.querySelector('[type="email"]').value;
    let password = e.target.querySelector('[type="password"]').value;
    api('/login', {post: {email, password}})
    .then(data => this.props.submit(data.token))
    .catch(data => this.setState({invalid: true}));
  }


}
