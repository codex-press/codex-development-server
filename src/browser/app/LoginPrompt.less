@import (reference) "../index.less";

.LoginPrompt {
  font-size: 1.2rem;
  width: 400px;
  height: auto;
  top: 50%;
  bottom: initial;
  transform: translateY(-50%);
  &:extend(.modal);

  h3 {
    font-size: 1.4rem;
    margin: 0;
  }

  label {
    display: block;
    margin: 1rem 0;
  }

  input {
    width: 100%;
    font-size: 1.5rem;
  }

  &.invalid input {
    color: red;
    outline: 1px solid red;
  }

  .locked {
    color: red;
  }

  .button-container {
    text-align: right;
  }

}


