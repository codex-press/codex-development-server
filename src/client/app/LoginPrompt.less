@import (reference) "../index.less";


.LoginPrompt {
  position: absolute;
  max-width: calc(100% - 2rem);
  padding: 1em;
  background: linear-gradient(rgba(30,30,30,0.95), rgba(0,0,0,0.9));
  font-size: 1.2rem;
  width: 400px;
  height: auto;
  top: 50%;
  left: 50%;
  bottom: initial;
  transform: translate(-50%, -50%);

  h3 {
    font-size: 1.4rem;
    margin: 0;
  }

  label {
    display: block;
    margin: 1rem 0;
  }

  input {
    width: 100%;
    font-size: 1.5rem;
  }

  &.invalid input {
    color: red;
    outline: 1px solid red;
  }

  .locked {
    color: red;
  }

  .button-container {
    text-align: right;
  }

}


