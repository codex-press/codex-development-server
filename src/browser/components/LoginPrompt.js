import React from 'react'

import { api } from '../utility'

import './LoginPrompt.less';

export default function LoginPrompt(props) {

  let formClass = 'LoginPrompt';
  if (props.status === 'rejected')
    formClass += ' invalid';

  const submit = e => {
    e.preventDefault();
    props.requestToken(
      e.target.querySelector('input[type="email"]').value,
      e.target.querySelector('input[type="password"]').value
    );
  }

  const cancel = e => {
    e.preventDefault();
    props.cancel();
  }

  let buttonClass = 'primary';
  if (props.status === 'pending')
    buttonClass += ' loading';

  return (
    <form className={ formClass } onSubmit ={ submit }>

      <h3>Codex Press Development Server</h3>

      { props.status === 'locked' &&
        <div className="locked">
          Your account is locked. Please wait and use the correct password.
        </div>
      }

      <label>
        Email
        <input type="email" />
      </label>

      <label>
        Password
        <input type="password" />
      </label>

      <div className="button-container">

        { props.cancel &&
          <button onClick={ cancel } >
            Cancel
          </button>
        }

        {' '}

        <button className={ buttonClass }>
          Login
        </button>

      </div>

    </form>
  );
}

