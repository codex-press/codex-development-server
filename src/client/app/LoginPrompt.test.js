import React from 'react'
import { shallow } from 'enzyme'
import { fromJS } from 'immutable'

import LoginPrompt from './LoginPrompt'



test('renders LoginPrompt', () => {
  const c = shallow(<LoginPrompt />)
  expect(c).toMatchSnapshot()
});



test('shows cancel button when provided a cancel fn as prop', () => {
  const c = shallow(<LoginPrompt cancel={ () => { } } />)
  expect(c).toMatchSnapshot()
});



test('can submit to requestToken', () => {
  const requestToken = jest.fn()
  const c = shallow(<LoginPrompt requestToken={ requestToken } />)
  const values = [ 'test@example.com', 'secret' ]
  c.find('form').simulate('submit', {
    preventDefault: () => {},
    target: { querySelector: () => ({ value: values.shift() }) },
  });
  expect(requestToken.mock.calls).toHaveLength(1)
  expect(requestToken.mock.calls[0]).toEqual([ 'test@example.com', 'secret' ])
});



test('shows loading indicator while token is pending', () => {
  const c = shallow(<LoginPrompt status="pending" />)
  expect(c.find('button')).toMatchSelector('.loading');
});



