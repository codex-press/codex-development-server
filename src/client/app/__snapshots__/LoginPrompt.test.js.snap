// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LoginPrompt 1`] = `
<form
  className="LoginPrompt"
  onSubmit={[Function]}
>
  <h3>
    Codex Press Development Server
  </h3>
  <label>
    Email
    <input
      type="email"
    />
  </label>
  <label>
    Password
    <input
      type="password"
    />
  </label>
  <div
    className="button-container"
  >
     
    <button
      className="primary"
    >
      Login
    </button>
  </div>
</form>
`;

exports[`shows cancel button when provided a cancel fn as prop 1`] = `
<form
  className="LoginPrompt"
  onSubmit={[Function]}
>
  <h3>
    Codex Press Development Server
  </h3>
  <label>
    Email
    <input
      type="email"
    />
  </label>
  <label>
    Password
    <input
      type="password"
    />
  </label>
  <div
    className="button-container"
  >
    <button
      onClick={[Function]}
    >
      Cancel
    </button>
     
    <button
      className="primary"
    >
      Login
    </button>
  </div>
</form>
`;
